### PR TITLE
Restored original header comments and copyrights. Relicensed to LGPL2.1

### PR DESCRIPTION
**Pull request by icdut24**
It has been brought to our attention that the header comments of the original gds2pov have been lost in our edits to GDS3D. We have corrected this in our original code and we would like to resolve this in your fork too. Moreover, we changed the license of our work to LGPL to make more future collaborations possible. We kindly ask you to also relicense your fork.

To reflect all changes, we have recompiled for all platforms, requiring some minor updates for Mac.

On behalf of the IC Design Group of the University of Twente,
Thomas

### CODE REVIEW

File: linux/Makefile
Comment (skuep) [inline linux/Makefile:1]: Why is the change in the Makefile needed?

File: mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D
Comment (skuep) [inline mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D:1]: Due to supply-chain attacks in the recent past (e.g. XZ project) I am a bit hesitant to upload new binaries to the repository. Is there any way to get a guarantee, that those are backdoor/malware free? Unfortunately I cannot compile the software for Windows or MacOS on my own.

File: mac/GDS3D.xcodeproj/project.xcworkspace/xcuserdata/anne-johanan.xcuserdatad/UserInterfaceState.xcuserstate
Comment (skuep) [inline mac/GDS3D.xcodeproj/project.xcworkspace/xcuserdata/anne-johanan.xcuserdatad/UserInterfaceState.xcuserstate:1]: There are quite a lot of new files in the Mac folders. Is that due to a version change in xcode? Maybe it makes sense to split this into a separate commit, since it doesn't have anything to do with the LICENSE. Although in this particular file, it looks to be some kind of user data of a person named Anne Johanan. It sounds a bit like this should not be in here.

File: mac/OpenGLView.mm
Comment (skuep) [inline mac/OpenGLView.mm:1]: Besides the changes in license text, I am unsure what the other changes are. I don't think they are LICENSE related

**Review by skuep — COMMENTED** [review]
Thanks for your PR, I have to be honest I was expecting only minor changes in the source code files in the context of LICENSE changes. I get that you probably want to put new binary blobs here, because of the changes in the command line outputs. 

However I am having a hard time replacing those binaries in my repository. Let me explain:
I want to be careful, since GDS3D is probably used in a lot of sensitive IC development environments and I can imagine a few current state actors wanting to get a backdoor into those environments (it happened in recent past like with the XZ project). Due to our E-mail conversation I am pretty sure that you are one of the original authors, but since your GitHub account has been created very recent, I am a bit wary. Don't take it personally! Unfortunately, due to limited time and availability of tools, I cannot produce or verify those new binaries for Windows & MacOS myself. 

Would this be okay for you:
a) Split this commit up, into pure changes of text files/comments, no added files, no actual code changes. Merge this into the codebase. Not sure what to do about the second portion of the split changes yet...
b) Add a paragraph to the top level README saying that the code license changed, but it is not (yet) reflected in the binaries, since they are the previous version. This would be fine with me, since it is (supposed to be) archived anyway, no one expects new binary releases. 
For up to date binaries, can I just link to your GitHub repository?  https://github.com/icdut24/GDS3D

Let me know how to proceed

File: linux/Makefile
Comment (icdut24) [reply linux/Makefile:1]: C++ 17 changed the behaviour of 'byte'. As such, the program only compiles on C++14 or C++11 (originally used). Due to those being old versions, it is needed to specify the version explicitely.

File: mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D
Comment (icdut24) [reply mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D:1]: I also don't have MacOS. Dr. Ir. Anne-Johan Annema of the ICD Group compiled it on MacOS 14 with -indeed- a new xcode.

File: mac/GDS3D.xcodeproj/project.xcworkspace/xcuserdata/anne-johanan.xcuserdatad/UserInterfaceState.xcuserstate
Comment (icdut24) [reply mac/GDS3D.xcodeproj/project.xcworkspace/xcuserdata/anne-johanan.xcuserdatad/UserInterfaceState.xcuserstate:1]: Yes, MacOS 14 with a recent version of xcode is used to compile this version. I might have overlooked these userfiles, I will delete them.

File: mac/OpenGLView.mm
Comment (icdut24) [reply mac/OpenGLView.mm:1]: Again, related to updated compilation requirements. The same changes are found in another fork: https://github.com/trilomix/GDS3D/blob/master/mac/OpenGLView.mm .

**Comment by icdut24** [comment]
I fully understand the problem with modified binary blobs. I'm happy you agree to correct our licensing. I will split the PR in a license part and an (up-to-date) compilation/compiled part if you want. I will leave it up to you if you want to merge the binaries or not. Compiled binaries are on our Github, but they are also updated on our original SourceForge website.

For archiving, I guess it would also be a possibility to only leave your modified scripts, and refer to the full source and binaries at our Github/SourceForge.

File: mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D
Comment (icdut24) [reply mac/GDS3D.app.dSYM/Contents/Resources/DWARF/GDS3D:1]: For Windows, also significant changes are needed for new compilers, but I succeeded in finding the 2010 one... Thus no changes there.

**Comment by skuep** [comment]
> I fully understand the problem with modified binary blobs. I'm happy you agree to correct our licensing. I will split the PR in a license part and an (up-to-date) compilation/compiled part if you want. I will leave it up to you if you want to merge the binaries or not. Compiled binaries are on our Github, but they are also updated on our original SourceForge website.
> 
> For archiving, I guess it would also be a possibility to only leave your modified scripts, and refer to the full source and binaries at our Github/SourceForge.

Thank you for understanding the issue. I think this is a good way forward. The license change is fine with me, most of the work has been done by you and my hat is off to that!
